File: src/main/kotlin/site/siredvin/lib/peripherals/owner/PeripheralOwnerAbility.java
package site.siredvin.lib.peripherals.owner;

@SuppressWarnings("InstantiationOfUtilityClass")
public class PeripheralOwnerAbility<T> {
    public static final PeripheralOwnerAbility<FuelAbility<?>> FUEL = new PeripheralOwnerAbility<FuelAbility<?>>();
    public static final PeripheralOwnerAbility<OperationAbility> OPERATION = new PeripheralOwnerAbility<OperationAbility>();

    public PeripheralOwnerAbility(){}
}

File: src/main/kotlin/site/siredvin/lib/peripherals/owner/PeripheralOwnerAbility.kt
package site.siredvin.lib.peripherals.owner

class PeripheralOwnerAbility<T> {
    companion object {
        val FUEL: PeripheralOwnerAbility<FuelAbility<*>> = PeripheralOwnerAbility()
        val OPERATION: PeripheralOwnerAbility<OperationAbility> = PeripheralOwnerAbility()
    }
}